core/common/src/main/java/alluxio/conf/RemovedKey.java

```diff
@@ -24,110 +24,144 @@
  * This class contains old {@link PropertyKey}s which have been removed from use
  *
  * This class is used to track keys which were deprecated in previous versions and subsequently
- * removed in a future version. We still keep them here so that it is possible to provide users
- * with useful information if they are found to be using an outdated property key.
+ * removed in a future version. We still keep them here so that it is possible to provide users with
+ * useful information if they are found to be using an outdated property key.
  *
  * Being removed and still used by an application denotes an error.
  *
  * @see InstancedConfiguration#validate()
  * @see PropertyKey#fromString(String)
  */
-public class RemovedKey {
+public final class RemovedKey {
 
-  private static final String V2_0_0_REMOVED = "this parameter is removed since v2.0.0.";
+  private static final String V2_0_0 = "v2.0.0";
+
+  /**
+   * @param version the version since which a property has been removed
+   *
+   * @return a warning message indicating a property has been removed since a given version
+   */
+  private static String removedSince(String version) {
+    return String.format("this property has been removed since %s.", version);
+  }
+
+  /**
+   * @param version the version since which a property has been removed
+   * @param newProperty the new property to use instead
+   * @return a warning message indicating a property has been removed since a given version
+   */
+  private static String replacedSince(String version, String newProperty) {
+    return String.format("this property has been removed since %s, use %s instead.", version,
+        newProperty);
+  }
 
   private static final Map<String, String> REMOVED_KEYS = new HashMap<String, String>(20) {
     {
-      put(Name.WEB_TEMP_PATH, "The alluxio web UI overhaul in v2.0 removed this parameter.");
```

Comment:
having another nested `Name` class is not necessary IMO, but adding more work. Without the `Name`s, the code of this class is much shorter.